packages/utils/dist/Signer.d.ts

```diff
@@ -1,8 +1,8 @@
 export class Signer {
     /**
      * @param {Buffer} pdfBuffer
-     * @returns {Buffer}
+     * @returns {Promise<Buffer> | Buffer}
      */
-    sign(pdfBuffer: Buffer): Buffer;
+    sign(pdfBuffer: Buffer): Promise<Buffer> | Buffer;
```

Comment:
I've missed it really. IMO the function has to be always async but I don't want to generalize unless we have some more Signrs. The node convention for this would be to have `signSync` if needed. I think this is fine with `Promise<Buffer> | Buffer` for the moment and we can tackle it with upcoming signers. Especially because all current implementations rely on it being synchronous and changing this forces a major release.

Reply:
My thoughts too. The underlying signer uses `await signer.sign()` so it can be async or sync and it works with `P12Signer` which is sync.

As you say, forcing it to be a promise will be a breaking change, but (on the other hand) these are just typings and there is no actual *requirement* for it to be a promise (as it stands). So we could make the types indicate a promise even though it wouldn't have to be?

Happy to take your lead on this one.